src/main/java/com/squareup/javapoet/ClassName.java

```diff
@@ -122,7 +122,7 @@ public static ClassName bestGuess(String classNameString) {
 
     // Add the package name, like "java.util.concurrent", or "" for no package.
     int p = 0;
-    while (p < classNameString.length() && Util.isLowerCase(classNameString.charAt(p))) {
+    while (p < classNameString.length() && Character.isLowerCase(classNameString.charAt(p))) {
```

Comment:
Definitely saw this in Eclipse back in the dark days of when I used Eclipse.

On Tue Jan 27 2015 at 8:54:59 AM Roman Mazur <notifications@github.com>
wrote:

> In src/main/java/com/squareup/javapoet/ClassName.java
> <https://github.com/square/javapoet/pull/208#discussion_r23622783>:
>
> > @@ -122,7 +122,7 @@ public static ClassName bestGuess(String classNameString) {
> >
> >      // Add the package name, like "java.util.concurrent", or "" for no package.
> >      int p = 0;
> > -    while (p < classNameString.length() && Util.isLowerCase(classNameString.charAt(p))) {
> > +    while (p < classNameString.length() && Character.isLowerCase(classNameString.charAt(p))) {
>
> A quick example https://ideone.com/68XtFO :)
> IMHO class loader can be not always guilty. A filesystem might have
> troubles with such file names, a bad collation might be set in a database,
> HTTP header with charset might be not respected...
> All this is getting better with time, but still it happens.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/square/javapoet/pull/208/files#r23622783>.
>